obspy/clients/filesystem/sds.py

```diff
@@ -169,24 +224,274 @@ def get_waveforms(self, network, station, location, channel, starttime,
+        :type location: str
+        :param location: Location code of requested data (e.g. "").
+        :type channel: str
+        :param channel: Channel code of requested data (e.g. "HHZ").
+        :type time: :class:`~obspy.core.utcdatetime.UTCDateTime`
+        :param time: Time of interest.
+        :type sds_type: str
+        :param sds_type: Override SDS data type identifier that was specified
+            during client initialization.
+        :rtype: str
+        """
+        sds_type = sds_type or self.sds_type
+        filename = self.FMTSTR.format(
+            network=network, station=station, location=location,
+            channel=channel, year=time.year, doy=time.julday, type=sds_type)
+        return os.path.join(self.sds_root, filename)
+
+    def get_availability_percentage(self, network, station, location, channel,
+                                    starttime, endtime, sds_type=None):
+        """
+        Get percentage of available data.
+
+        :type network: str
+        :param network: Network code of requested data (e.g. "IU").
+        :type station: str
+        :param station: Station code of requested data (e.g. "ANMO").
+        :type location: str
+        :param location: Location code of requested data (e.g. "").
+        :type channel: str
+        :param channel: Channel code of requested data (e.g. "HHZ").
+        :type starttime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+        :param starttime: Start of requested time window.
+        :type endtime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+        :param endtime: End of requested time window.
+        :type sds_type: str
+        :param sds_type: Override SDS data type identifier that was specified
+            during client initialization.
+        :rtype: 2-tuple (float, int)
+        :returns: 2-tuple of percentage of available data (``0.0`` to ``1.0``)
+            and number of gaps/overlaps.
+        """
+        if starttime >= endtime:
+            msg = ("'endtime' must be after 'starttime'.")
+            raise ValueError(msg)
+        sds_type = sds_type or self.sds_type
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore", "Keyword headonly cannot be combined with "
+                "starttime, endtime or dtype.", UserWarning)
+            st = self.get_waveforms(network, station, location, channel,
+                                    starttime, endtime, sds_type=sds_type,
+                                    headonly=True, _no_trim_or_merge=True)
+        st.sort(keys=['starttime', 'endtime'])
+        st.traces = [tr for tr in st
+                     if not (tr.stats.endtime < starttime or
+                             tr.stats.starttime > endtime)]
+
+        if not st:
+            return (0, 1)
+
+        total_duration = endtime - starttime
+        # sum up gaps in the middle
+        gaps = [gap[6] for gap in st.getGaps()]
```

Comment:
Not *yet*.. I put a note in there https://github.com/obspy/obspy/issues/1039#issuecomment-150163012 already, so we do it before release..